scripts/crowdloan-rewards/common/generateVestings.ts

```diff
@@ -0,0 +1,129 @@
+var assert = require('assert');
+import BigNumber from "bignumber.js"
+import { log } from './performVestings'
+
+const HDX = new BigNumber(1_000_000_000_000);
+
+export type RewardsData = {
+    address: string,
+    totalRewards: string
+}
+
+
+export type DynamicVestingInfo = {
+  destination: string,
+  schedule: {
+      amountToBeVested: string,
+      start: string,
+      period: string,
+      per_period: string,
+      period_count: string 
+  }
+}
+
+export const generateVestingsAndWriteToFs = function(
+    rewards: RewardsData[],
+    startBlock: string,
+    endBlock: string,
+    triple: boolean,
+    prefix: string
+) {
+    log(`Generating vesting schedules for ${prefix} crowdloaners`);
+
+    let totalRewards = new BigNumber(0);
+
+    let vestingsData = rewards.flatMap(rewardData => {
+        let rewards = new BigNumber(rewardData.totalRewards);
+        rewards = triple ? rewards.multipliedBy(3) : rewards;
+
+        totalRewards = totalRewards.plus(rewards);
```

Comment:
no sideefects pls, just reduce vestingsData after they are generated

this doesn't even guarantee you're really rewarding this as it relies on generateVesting to be correct, you should sum it up after

you already have `calculateTotalVestedRewards`, use it

if the goal is to count total rewards in `rewards` before transformation, sum it up before, not during it, but you should compare those two anyway probably